packages/analytics-js-plugins/src/deviceModeDestinations/index.ts

```diff
@@ -72,51 +78,44 @@ const DeviceModeDestinations = (): ExtensionPlugin => ({
     ) {
       const integrationsCDNPath = state.lifecycle.integrationsCDNPath.value;
       const activeDestinations = state.nativeDestinations.activeDestinations.value;
-      const onLoadCallback =
-        externalScriptOnLoad ??
-        ((id?: string) => {
-          if (!id) {
-            return;
-          }
-
-          logger?.debug(`Destination script with id: ${id} loaded successfully`);
-        });
 
       activeDestinations.forEach(dest => {
-        const sdkName = destDispNamesToFileNamesMap[dest.displayName];
+        const sdkName = destDisplayNamesToFileNamesMap[dest.displayName];
         const destSDKIdentifier = `${sdkName}_RS`; // this is the name of the object loaded on the window
-        logger?.debug(`Loading destination: ${dest.userFriendlyId}`);
 
+        let timeoutId: number;
+        let intervalId: number;
         if (!isDestinationSDKEvaluated(destSDKIdentifier, sdkName, logger)) {
           const destSdkURL = `${integrationsCDNPath}/${sdkName}.min.js`;
-          externalSrcLoader
-            .loadJSFile({
-              url: destSdkURL,
-              id: dest.userFriendlyId,
-              callback: onLoadCallback,
-            })
-            .catch(e => {
```

Comment:
do errors get handled internally in externalSrcLoader loadJSFile?

Reply:
Yes.